### PR TITLE
Feature/refactor reader tests

### PR DESCRIPTION
**Pull request by ShimonSte**
# Add comprehensive read/write test coverage for Spark 3.3, 3.4, and 3.5

## Overview

This PR adds comprehensive test coverage for ClickHouse Spark connector read and write operations across Spark 3.3, 3.4, and 3.5.

### Read Tests (Commit 1: 7737fdd)
- **ClickHouseReaderTestBase.scala**: Shared test trait with 48 test scenarios
- **ClickHouseBinaryReaderSuite.scala**: Binary format configuration
- **ClickHouseJsonReaderSuite.scala**: JSON format configuration

### Write Tests (Commit 2: 2bbbb52)
- **ClickHouseWriterTestBase.scala**: Shared test trait with 17 test scenarios
- **ClickHouseJsonWriterSuite.scala**: JSON format configuration
- **ClickHouseArrowWriterSuite.scala**: Arrow format configuration

### Read Tests (288 total across all Spark versions)
- **48 test scenarios** per Spark version
- **Binary format**: 47/48 passing per version
- **JSON format**: 47/48 passing per version
- **Pass rate**: 98% (94/96 per version)

### Write Tests (102 total across all Spark versions)
- **17 test scenarios** per Spark version
- **JSON format**: 17/17 passing per version (100%)
- **Arrow format**: 17/17 passing per version (100%)
- **Pass rate**: 100% (34/34 per version)

### Overall Statistics
- **Total Tests**: 390 (130 per Spark version)
- **Passing**: 384 (98.5%)
- **Failing**: 6 (known bugs with fixes on separate branches)

### Read Tests Cover:
- All primitive types (Boolean, Byte, Short, Int, Long, Float, Double)
- Large integers (UInt64, Int128, UInt128, Int256, UInt256)
- Decimals (Decimal32, Decimal64, Decimal128)
- Date/Time types (Date, Date32, DateTime, DateTime32, DateTime64)
- String types (String, UUID, FixedString)
- Enums (Enum8, Enum16)
- IP addresses (IPv4, IPv6)
- JSON data
- Collections (Arrays, Maps)
- Edge cases (empty strings, long strings, empty arrays, nullable variants)

### Write Tests Cover:
- Primitive types (Boolean, Byte, Short, Int, Long, Float, Double)
- Decimal types
- String types (regular and empty)
- Date and Timestamp types
- Collections (Arrays and Maps, including empty variants)
- Nullable variants

### Failing Tests (6 total, 2 per Spark version)
1. **FixedString (JSON read)** - Base64 encoding issue
   - Fixed on branch: `fix/json-reader-fixedstring`
2. **Empty Map (Binary read)** - NullPointerException
   - Fixed on branch: `fix/binary-reader-empty-map`

### Shared Trait Pattern
Tests use a shared trait pattern for code reuse:
- Base traits contain all test logic
- Suite classes only configure format settings
- Easy to add new formats by extending the trait

### CODE REVIEW

File: spark-3.3/clickhouse-spark-it/src/test/scala/org/apache/spark/sql/clickhouse/single/ClickHouseBinaryReaderSuite.scala [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * Test suite for ClickHouse Binary Reader.
+ * Uses binary format for reading data from ClickHouse.
+ * All test cases are inherited from ClickHouseReaderTestBase.
+ */
+abstract class ClickHouseBinaryReaderSuite extends ClickHouseReaderTestBase {
+  
+  // Override to use binary format instead of JSON
+  override protected def sparkConf: SparkConf = super.sparkConf
+    .set("spark.clickhouse.read.format", "binary")
+    .set("spark.clickhouse.write.format", "binary")
```
Comment (mzitnik) [inline spark-3.3/clickhouse-spark-it/src/test/scala/org/apache/spark/sql/clickhouse/single/ClickHouseBinaryReaderSuite.scala]: I do not think we can write in binary. please check

File: spark-3.3/clickhouse-spark-it/src/test/scala/org/apache/spark/sql/clickhouse/single/ClickHouseReaderTestBase.scala [hunk 2]
```diff
@@ -0,0 +1,959 @@
...
+      checkAnswer(
+        df,
+        Row(1, -128.toByte) :: Row(2, null) :: Row(3, 127.toByte) :: Nil
+      )
+    }
+  }
+
+  // Test: ShortType decoding
+  test("decode ShortType - min and max values") {
+    withKVTable("test_db", "test_short", valueColDef = "Int16") {
```
Comment (mzitnik) [inline spark-3.3/clickhouse-spark-it/src/test/scala/org/apache/spark/sql/clickhouse/single/ClickHouseReaderTestBase.scala]: Should we have Nullable Int16?

File: spark-3.3/clickhouse-spark-it/src/test/scala/org/apache/spark/sql/clickhouse/single/ClickHouseWriterTestBase.scala [hunk 3]
```diff
@@ -0,0 +1,431 @@
...
+      assert(result.length == 3)
+      // Boolean is stored as UInt8 in ClickHouse, reads back as Short
+      assert(result(0).getShort(1) == 1)
+      assert(result(1).isNullAt(1))
+      assert(result(2).getShort(1) == 0)
+    }
+  }
+
+  // Test: ByteType writing
+  test("write ByteType - min and max values") {
```
Comment (mzitnik) [inline spark-3.3/clickhouse-spark-it/src/test/scala/org/apache/spark/sql/clickhouse/single/ClickHouseWriterTestBase.scala:75]: We need to add corresponding tests for nullable with Byte type.

Comment (ShimonSte) [reply spark-3.3/clickhouse-spark-it/src/test/scala/org/apache/spark/sql/clickhouse/single/ClickHouseWriterTestBase.scala:75]: We already have one

File: spark-3.3/clickhouse-spark-it/src/test/scala/org/apache/spark/sql/clickhouse/single/ClickHouseReaderTestBase.scala (hunk 2 above)
Comment (ShimonSte) [reply spark-3.3/clickhouse-spark-it/src/test/scala/org/apache/spark/sql/clickhouse/single/ClickHouseReaderTestBase.scala]: I added new tests
